File: MyMusicListBackend/MyMusicList/MyMusicList/Models/Song.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace MyMusicList.Models
{
    public class Song
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string NameOfSong { get; set; }
        [Required]
        public string NameOfSinger { get; set; }
        [Required]
        [Url]
        public string UrlOfSong { get; set; }
        [Required]
        [Range(1, 5)]
        public int SongRating { get; set; }
        [Required]
        public bool Favorite { get; set; }
        [Required]
        public DateTime SongEntryDate { get; set; }
        public DateTime SongLastEdit { get; set; }
        public Category Category { get; set; }
        public int CategoryId { get; set; }
    }
}

File: MyMusicList/src/app/models/Song.ts
import { Category } from "./Category";

export class Song {
    id: string;
    nameOfSong: string;
    nameOfSinger: string;
    favorite: string;
    enterDate: Date;
    urlOfSong: string;
    lastEdited: Date;  
    category: Category;
    songRating: string;
    
    constructor(id: string,
        nameOfSong: string,
    nameOfSinger: string,
    favorite: string,
    enterDate: Date,
    url: string,
    lastEdited: Date, 
    category: Category,
    songRating: string) {
      this.id = id;
      this.nameOfSong=nameOfSong;
      this.nameOfSinger=nameOfSinger;
      this.favorite = favorite;
      this.enterDate=enterDate;
      this.urlOfSong=url;
      this.lastEdited=lastEdited;
      this.songRating=songRating;
      this.category=category;
    }
  
  }
  